vm/actor/src/builtin/miner/state.rs

```diff
@@ -191,13 +189,13 @@ impl State {
         }
     }
     pub fn compute_proving_set<BS: BlockStore>(
-        &self,
+        &mut self,
```

Comment:
are any of these methods changing an internal state?

Reply:
Okay so here's the thing with these, it does need mutable reference, but only because I don't want to duplicate decoding and checking on the bitfield, since it happens so many times. 

It is an unfortunate side effect and I'm very open to alternatives (just have to keep in mind for this it must be very efficient as it could almost halt the node in very specific cases if not). The mutation does not change the actual value, just changes the data from being RLE encoded to decoded and clears the cache

Reply:
This sounds good to me